### ISSUE TITLE
[Goonj] Moving Staging org to UAT in Prod

### ISSUE THREAD

**Issue by utkarshavni** [avniproject/Goonj#63]
Reference sheet:
https://docs.google.com/spreadsheets/d/1qs1f-hjXeReuC-wg6R3780wik9omW9eofsjI8lGjk5E/edit

Steps:
- [x] Download bundle without locations
- [x] Upload bundle without locations
- [x] Create dummy locations
- [x] Create Catchments and users
- [x] Configure UserGroups (Dashboards and permissions)
- [x] Void dashboards and related entities which were actually voided in Staging env (do same for report_cards and dashboard_sections and dashboard_section_mappings)
- [x] Configure syncConceptValues for users based on Shivangi's requirements
- [x] Create Goonj Avni api and salesforce system users
- [x] Configure new credentials (Prod Avni Goonj Api user and password.Salesforce Prod system credentials.Tasks to be run and their frequency)
- [x] Migrate the goonj Int metadata mapping info from Staging to Prod integration DB -> filter and retain only entries from- Goonj Integration system in the upload sql file
```
1. Take dump of staging avni-int db
  - Delete unwanted table entries
  - Delete non-goonj org entries
  - Delete goonj transactional data
  - Replace hardcoded ids with corresponding ids (query)

2. Then in prod avni-int db
  - Delete goonj data (transactional and meta data)
  - Apply dump
```
- [x] Deploy latest integration code to staging int server
- [x] Setup integration_system_config table entries and then restart Goonj service
- [x] Setup Integration system Monitoring and alert. Configure Errors /  Exceptions to be ignored.
- [x] Goonj Staging integration should start syncing between Goonj Salesforce Partial environment and the Avni Goonj UAT prod org api user. Sanity testing using Prod APK for Goonj prod implementation.

~- [ ] Upload locations for Maharashtra - async operation~

**Comment by utkarshavni** [comment]
Approx time for task - 24-28 hours.

**Comment by himeshr** [comment]
## Upload bundle without locations
### ISSUES

- [x] Unrecognized field "lastNameOptional" 
- [x] Validation failed for classes [org.avni.server.domain.OperationalSubjectType] during persist time for groups [javax.validation.groups.Default, ]
List of constraint violations:[
	ConstraintViolationImpl{interpolatedMessage='must not be null', propertyPath=subjectType, rootBeanClass=class org.avni.server.domain.OperationalSubjectType, messageTemplate='{javax.validation.constraints.NotNull.message}'}
] : Modify operationalSubjectTypes.json 

- [x] could not execute statement; SQL [n/a]; constraint [subject_type_check]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement
org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:259) : formMapping.json

- [x] Validation failed for classes [org.avni.server.domain.GroupPrivilege] during persist time for groups [javax.validation.groups.Default, ]
List of constraint violations:[
	ConstraintViolationImpl{interpolatedMessage='must not be null', propertyPath=privilege, rootBeanClass=class org.avni.server.domain.GroupPrivilege, messageTemplate='{javax.validation.constraints.NotNull.message}'}
] : groupPrivilige.json 

## Fixes:
 - Remove the column from subjectTypes.json (all header and all other data rows)
 - Remove entries from groupPrivilege.json where => "privilegeUUID" : "fe6e5550-26f7-4c2f-b18a-b24af00d7cde",

**Comment by BEULAHEVANJALIN** [comment]
- [x] Take dump of staging avni-int db
- [x] Delete unwanted table entries
- [x] Delete non-goonj org entries
- [x] Delete goonj transactional data 
- [x] Replace hardcoded ids with corresponding ids (query)

Then in prod avni-int db
- [x] Delete goonj data (transactional and meta data)
- [x] Apply dump

**Comment by BEULAHEVANJALIN** [comment]
Draft dump scripts:

[https://gist.github.com/BEULAHEVANJALIN/2f71d2030e117da7987b4cb0ccad3ab1](https://gist.github.com/BEULAHEVANJALIN/2f71d2030e117da7987b4cb0ccad3ab1)

**Comment by BEULAHEVANJALIN** [comment]
Delete duplicate dashboard sections

```
select *
from dashboard_section_card_mapping
where uuid in ('493724fe-d5f8-44a5-8d99-3f9cfe3a7518',
               '58400a31-de48-4090-aabd-e628e9309d64',
               '59d8d13a-57b7-494b-a7e3-e39754b94283',
               'e72d043b-5a30-4993-b153-449cf2781347',
               'd1f02abf-ef85-4773-8e61-c3e3b70100c2');

delete
from dashboard_section_card_mapping
where uuid in ('493724fe-d5f8-44a5-8d99-3f9cfe3a7518',
               '58400a31-de48-4090-aabd-e628e9309d64',
               '59d8d13a-57b7-494b-a7e3-e39754b94283',
               'e72d043b-5a30-4993-b153-449cf2781347',
               'd1f02abf-ef85-4773-8e61-c3e3b70100c2');
```

**Comment by BEULAHEVANJALIN** [comment]
Bundle uploaded on Jul 20, 2023

[https://app.zenhub.com/files/498306799/d4079abd-374d-4c45-93fc-4219600d11f8/download](https://app.zenhub.com/files/498306799/d4079abd-374d-4c45-93fc-4219600d11f8/download)

**Comment by BEULAHEVANJALIN** [comment]
User & Catchment zip file
[https://app.zenhub.com/files/498306799/05e68446-de55-464c-9d2f-fb447362275e/download](https://app.zenhub.com/files/498306799/05e68446-de55-464c-9d2f-fb447362275e/download)

**Comment by himeshr** [comment]
Uploaded Locations:
- Specific district locations
- Other blocks and villages

[https://app.zenhub.com/files/498306799/70b04cc2-6b8d-4270-ab5f-3a1a1322f93d/download](https://app.zenhub.com/files/498306799/70b04cc2-6b8d-4270-ab5f-3a1a1322f93d/download)

**Comment by Dinesh2019** [comment]
- [x] We are facing the issue of rule failure when we select the inventory and activity for creating the distribution in the client app 
- [x] After completing the dispatch receipt the inventory is not created in the sales force (Beulah is working on it)

**Comment by Dinesh2019** [comment]
Test Scearios worked on 
1.Created the demand with kit and non-kit material 
2.created the single demand with both kit and non-kit material 
3.validated the status of the demand from pending to under process 
4.create the dispatch with kit and non-kit
5created the single dispatch with both kit and non kit
6.created the dispatch with same kit type material but with different sub-type
7.complete the dispatch receipt with quantity as yes 
8.complete the dispatch receipt with quantity as no 
9.validate the inventory created for kit type in salesforce once the dispatch receipt is completed 
10.validate the inventory send to Avni-client app.
11.Create the distribution with different types of initiative
12.Validate the source id is added
13.validate the other is added in block and village 
14.validate the tola/mohalla is added 
15.validate the distrubtion date cannot be is future
16.validate the account name is displayed 
17.validate the initiative is need as CFW-Rahat and Rahat is should display the type of disaster
18.validate the s2s and cfw-s2s is selected it should display the type of school
19.validate the inventory is selected and it display the unit initial quantity and current quantity
20.validate the same inventory is not selected if we selected it should display the error 
21.validate the quaninty is not greater than current quantity 
22.validate the is the activity available for linking as yes or no
23.validate the Activity is selected and it display the Number of participants 
24.validate the same Activity is not selected if we selected it should display the error 
25validate the number of persons is not greater than total participants.
26.validate the distrubtion is saved and should displayed in the salesforce
27.create the activity with CFW NJPC and S2S
28.validate the source id is displayed 
29.validate the other in block and village
30.validate the tool/mohalla 
31.validate the account name is displayed
32.validate the start date and end date cannot be in future 
33.validate the no of working days is between the start and end date
34.save the activity and validate the it is displayed in the sales force.
35.validated the dashboard according to scenarios.
36.valid the transaltion
